pkg/object/httpserver/spec.go

```diff
@@ -211,3 +221,15 @@ func (p *Path) Validate() error {
 
 	return nil
 }
+
+func (q *Query) initQueryRoute() {
+	q.queryRE = regexp.MustCompile(q.Regexp)
```

Comment:
when the regexp is an empty string, we don't need to compile it, this could save some memory.